typescript-test-samples/lambda-handler-dynamodb/README.md

```diff
@@ -0,0 +1,121 @@
+
+In this sample, we have used AWS CDK (Cloud Development Kit) for infrastructure as code but the pattern can be also be used with a variety of systems including SAM, Serverless Framework, CloudFormation and Terraform. This pattern simplifies the process of attaching a debugger to the handler code. You may generate synthetic events using several methods including the ‘sam local generate-event’ command, by copying event template samples in the AWS Lambda console test section, or by printing events to CloudWatch logs.
+
+![Test Pattern](img/pattern_02_lambda_handler_test.png)
+
+### Limitations
+
+The IAM policies contained in your local environment variables may differ from those contained in the execution role attached to the Lambda function in the AWS cloud environment. Therefore this pattern may not accurately test IAM permissions. You may mitigate this risk by using the [IAM Roles Anywhere](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/introduction.html) service to obtain temporary security credentials for the same execution role that the Lambda function uses in the cloud context. This service is open source and you may [view the code on GitHub](https://github.com/aws/rolesanywhere-credential-helper).
+
+[Top](#contents)
+---
+
+## About this Example
+
+This example contains an [AWS Lambda](https://aws.amazon.com/lambda/) function and [Amazon DynamoDB](https://aws.amazon.com/dynamodb/) table as core resources.
+
+AWS CDK is used to deploy the AWS Lambda function and Amazon DynamoDB table. The AWS Lambda function generates an item in the JSON format and writes this item to the DynamoDB table using the AWS SDK. The DynamoDB table item is a JSON object with format:
+
+```json
+{
+    "PK": "string",
+    "SK": "string"
+}
+```
+
+### Key Files in the Project
+
+  - [app.ts](src/app.ts) - Lambda handler code to test
+  - [lambda-handler-dynamodb-stack.ts](lib/lambda-handler-dynamodb-stack.ts) - CDK stack for deploying required resources
+  - [integ-handler.test.ts](test/integration/integ-handler.test.ts) - Integration tests on a live stack
+
+[Top](#contents)
+---
+
+## Integration Test
+
+### Integration Test Description
+
+In order to run integration tests in the cloud we will first need to deploy the full CDK stack which will create a Lambda function and DynamoDB table in your AWS account. The integration test invokes the handler function method on the local machine, passing it a synthetic event as a parameter. The handler function executes within your local environment. When the SDK writes to DynamoDB, it uses the AWS credentials contained in your local environment.
+
+### Run the Integration Tests
+
+[integ-handler.test.ts](test/integration/integ-handler.test.ts) 
+
+Deploy the full stack:
+
+```shell
+lambda-handler-dynamodb$ npm install
+lambda-handler-dynamodb$ cdk deploy
```

Comment:
I get an error trying to run this. Looks like the `/lib` and/or `/bin` directory is missing from the commit. Its likely it was ignored by the `.gitignore`. You may have to explicitly tell git to add it.

For example:
`git add lib/lambda-handler-dynamic-stack.ts`

Reply:
Yeah the gitignore in the main repo ignored lib/ file so manually added in this pattern. You should be able to run the cdk app now.

Reply:
Still getting an error here unfortunately. Might need the `*.ts* file from under `/bin` added to the repo too.